axum-login/src/redis_store.rs

```diff
@@ -0,0 +1,81 @@
+use std::marker::{PhantomData, Unpin};
+
+use async_trait::async_trait;
+#[cfg(feature = "redis")]
+use redis::{Client, Commands, Connection, FromRedisValue};
+
+use crate::{user_store::UserStore, AuthUser};
+
+/// Redis user store.
+#[derive(Clone, Debug)]
+pub struct RedisStore<User, Role = ()> {
+    client: Client,
+    _user_type: PhantomData<User>,
+    _role_type: PhantomData<Role>,
+}
+
+impl<User, Role> RedisStore<User, Role> {
+    /// Creates a new store with the provided client.
+    pub fn new(client: Client) -> Self {
+        Self {
+            client,
+            _user_type: Default::default(),
+            _role_type: Default::default(),
+        }
+    }
+}
+
+#[async_trait]
+impl<UserId, User, Role> UserStore<UserId, Role> for RedisStore<User, Role>
+where
+    UserId: Sync + redis::ToRedisArgs,
+    Role: PartialOrd + PartialEq + Clone + Send + Sync + 'static,
+    User: AuthUser<UserId, Role> + Unpin + FromRedisValue,
+{
+    type User = User;
+
+    /// user_id is the unique Redis key under which the user data is stored.
+    ///
+    /// Note: This key must be returned by your implementation of `get_id(...)`
+    /// for the `AuthUser`-trait.
+    async fn load_user(&self, user_id: &UserId) -> crate::Result<Option<Self::User>> {
+        let mut con: Connection = self.client.get_connection()?;
+        let user: Option<User> = con.get(user_id).ok();
```

Comment:
Remove the `.ok()`, replace with `con.get(user_id)?;` or better, change the whole thing into:
```rust
con.get::<_, Option<Self::User>>(user_id).wrap_err("Error when retrieving user from redis")
```